src/core/cgroup.c

```diff
@@ -4398,6 +4398,8 @@ int cgroup_runtime_deserialize_one(Unit *u, const char *key, const char *value,
                 if (r < 0)
                         log_unit_debug_errno(u, r, "Failed to set cgroup path %s, ignoring: %m", value);
 
+                (void) unit_watch_cgroup(u);
+                (void) unit_watch_cgroup_memory(u);
```

Comment:
So, I can see how this fixes the issue at hand, but the root cause is going to remain unresolved: after unit deserialization cgroups are no longer properly re-realized.

As I explained in the commit message, unit_invalidate_cgroup() should have enqueued all cgroups we care about for re-realization...

Reply:
Given this fixes the issue, _and_ pushes the CI runtimes down to reasonable values again, can we get this merged and file an issue to track the root cause, please?

Reply:
I have not tested it manually, but you can see from the CI, it's back to ~30 minutes for the full run, while on main it's like 2 hours, due to all the delays as the user sessions get stuck